pandas/core/series.py

```diff
@@ -2864,35 +2867,46 @@ def map(self, arg, na_action=None):
         1    foo
         2    bar
         3    baz
+        dtype: object
 
         >>> x.map(y)
         one   foo
         two   bar
         three baz
+        dtype: object
+
+        Map a function to a :class:`pandas.Series`.
 
-        If `arg` is a dictionary, return a new Series with values converted
-        according to the dictionary's mapping:
+        >>> x.map(lambda x: x**2)
+        one      1
+        two      4
+        three    9
+        dtype: int64
```

Comment:
I would start by the example using a function as a parameter, which I think it's a more common use than with a series.